src/bartiq/transform.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2024 PsiQuantum, Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import copy
+from typing import Any, Dict, Union
+
+import sympy
+from qref import SchemaV1
+
+def add_aggregated_resources(
+    aggregation_dict: Dict[str, Dict[str, Any]], qref_obj: Union[SchemaV1, dict]
+) -> Union[SchemaV1, dict]:
+    """
+    Add aggregated resources to qref program based on the aggregation dictionary.
```

Comment:
Please add an example of what does the structure of the `aggregation_dict` looks like in the docstring.